public/js/pimcore/object/tags/block.js

```diff
@@ -311,6 +311,25 @@ pimcore.object.tags.block = Class.create(pimcore.object.tags.abstract, {
 
         this.dataFields = {};
         this.currentData = {};
+
+        this.updateBlockIndices();
+    },
+
+    updateBlockIndices: function() {
+        for (let itemIndex = 0; itemIndex < this.component.items.items.length; itemIndex++) {
+            let item = this.component.items.items[itemIndex];
+
+            for (let j = 0; j < this.currentElements.length; j++) {
+                if (item !== this.currentElements[j].container) continue;
+
+                let fields = this.currentElements[j].fields;
+                for (let fieldName in fields) {
+                    if (this.currentElements[j].fields.hasOwnProperty(fieldName)) {
```

Comment:
@procnet0 Sorry, I don't get why we need this check, because for me `this.currentElemenst[j].fields` is the same as `fields` and you iterate over fields. So the prop `fieldName` should always be in `fields` right?